components/private/State/src/MDCControlState.swift

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/**
+ MDCControlState represents all Material states, including the special interactive
+ state, which is any state that is not normal, disabled or error.
+ * Eight distinctive states are supported.
+ * Material's enabled and pressed states are renamed as their UIControlState state names (which
+   is normal and highlighted).
+ * We are still experimenting with state combinations. Whther adding names for state combinations
+   (like: selectedHighlighted) is needed, is stil in debate.
+ */
+struct MDCControlState: OptionSet, CustomStringConvertible {
```

Comment:
Whatever we ultimately merge in won't be in Swift though right?

Reply:
The public ColorScheme API will continue to be in objective C. Examples will remain in Swift. I'm not sure yet about non-public API.

Reply:
Would an Objective-C MDCControlState be an NS_Options? Or an NSObject?

Reply:
All implementation will be in objective C, at least until swift 5 is out, to maintain backward compatibility.  MDCControlState will be written in objective C, and is going to be an option-set/bit-mask. See #6585.